rust_snuba/src/processors/utils.rs

```diff
@@ -39,3 +40,38 @@ where
     };
     Ok(seconds_since_epoch.timestamp() as u32)
 }
+
+#[allow(dead_code)]
+pub struct CogsAccountant {
+    accountant: UsageAccountant<KafkaProducer>,
+}
+
+impl CogsAccountant {
+    #[allow(dead_code)]
+    fn new(bootstrap_servers: &str) -> Self {
+        let config = KafkaConfig::new_producer_config(bootstrap_servers, None);
+        Self {
+            accountant: UsageAccountant::new_with_kafka(config, None, None),
+        }
+    }
+
+    fn record_bytes(&mut self, resource_id: &str, app_feature: &str, amount_bytes: u64) {
```

Comment:
how will this be used -- is it going to be per-batch in snuba, or per-message?

I feel that batches are fairly homogenous when it comes to COGS and it should be possible to call it once per batch.

if it is per-message, perhaps we should attach COGS to `BytesInsertBatch` and only record them in clickhouse.rs? that would inherently be more performant and also address filippo's logging concern

Reply:
Per message - the accountant library takes care of batching and flushing periodically so calling record is really cheap.

Reply:
though actually the library does call `Utc::now()` on every message so might still affect perf